spinn_front_end_common/utilities/database/database_writer.py

```diff
@@ -267,6 +270,21 @@ def create_atom_to_event_id_mapping(self, machine_vertices):
                     """, ((m_vertex_id, int(key), i) for i, key in atom_keys)
                 )
 
+    def _is_machine_lpg(self, vertex):
+        return (isinstance(vertex, LivePacketGatherMachineVertex) and
+                not isinstance(vertex.app_vertex, LivePacketGather))
+
+    def _is_machine(self, edge):
+        return (isinstance(edge.pre_vertex, AbstractOneAppOneMachineVertex) and
+                isinstance(edge.post_vertex, AbstractOneAppOneMachineVertex))
+
+    def _get_machine_lpg_mappings(self, part):
```

Comment:
I think this will read a lot cleaner if the three functions are combined.

It also avoids directly doing edge.post_vertex.machine_vertex..app_vertex

Reply:
Done